### PR TITLE
Commerce: Seeding Frontend

### PR DESCRIPTION
**Pull request by zfox23**
DNM until corresponding backend changes are merged. Currently testable on Staging.

The code in this PR does the following:
1. If there are no transactions in the model populated from the `inventory()` endpoint response, we display a `this is a closed beta` message on Wallet Home (see the image below).
2. Removes machine fingerprint from receiveAt transaction

The backend will implement the following:
1. A new table with a list of usernames associated with a "alreadySeeded" `bool` defaulting to `false`
2. Every time something hits the `receive_at`, `balance`, or `inventory` backend endpoints, the backend will check that table to determine if a user should be seeded, and then seed their Wallet if necessary.

image

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/4768/HighFidelity-Beta-PR11771-39e8a3925e5a191a5ceb9c98225d925a536a8aea.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/4775/HighFidelity-Beta-PR11771-38fde43009e4221030b89285791f9e397d6f2548.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/4775/HighFidelity-Beta-PR11771-38fde43009e4221030b89285791f9e397d6f2548.exe

File: interface/src/commerce/Ledger.cpp [hunk 1]
```diff
@@ -90,7 +90,7 @@ void Ledger::buy(const QString& hfc_key, int cost, const QString& asset_id, cons
     signedSend("transaction", transactionString, hfc_key, "buy", "buySuccess", "buyFailure", controlled_failure);
 }
 
-bool Ledger::receiveAt(const QString& hfc_key, const QString& old_key, const QString& machine_fingerprint) {
+bool Ledger::receiveAt(const QString& hfc_key, const QString& old_key) {
```
Comment (davidkelly) [inline interface/src/commerce/Ledger.cpp]: 👍

File: interface/src/commerce/Ledger.cpp [hunk 2]
```diff
@@ -99,13 +99,7 @@ bool Ledger::receiveAt(const QString& hfc_key, const QString& old_key, const QSt
...
     }
 
-    QJsonObject transaction;
-    transaction["hfc_key"] = hfc_key;
-    transaction["machine_fingerprint"] = machine_fingerprint;
-    QJsonDocument transactionDoc{ transaction };
-    auto transactionString = transactionDoc.toJson(QJsonDocument::Compact);
-
-    signedSend("transaction", transactionString, old_key, "receive_at", "receiveAtSuccess", "receiveAtFailure");
+    signedSend("public_key", hfc_key.toUtf8(), old_key, "receive_at", "receiveAtSuccess", "receiveAtFailure");
```
Comment (davidkelly) [inline interface/src/commerce/Ledger.cpp]: 👍

**Comment by davidkelly** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/4801/HighFidelity-Beta-PR11771-19b7c6e025113eceeb40d2b1fbb1ce39543a011e.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/4801/HighFidelity-Beta-PR11771-19b7c6e025113eceeb40d2b1fbb1ce39543a011e.exe

**Comment by davidkelly** [comment]
Test Plan:

Login with an account that isn't on the list of people in the beta
enable commerce (`"commerce": true`, in AppData/Roaming/<this pr>/Interface.json)
create wallet
Note that it tells you about the closed beta
Go to help in the wallet
reset the wallet
setup wallet again
still should tell you about closed beta
have your user name placed on the list (ask me, zach, or howard)
after your username is on the list, just open the wallet
you should see 10,000 hfc
reset your wallet again
you should not get another 10K HFC
🍻

**Comment by YuppyPlays** [comment]
Testing.

**Comment by YuppyPlays** [comment]
Testing complete:

Login with an account that isn't on the list of people in the beta **Check**
enable commerce ("commerce": true, in AppData/Roaming//Interface.json) **Check**
create wallet
Note that it tells you about the closed beta **Check, though not sure if this is intentional, the text appears to appear for 4 or so seconds, then disappears for half a second, and then continues that loop.**
Go to help in the wallet **Check**
reset the wallet **Check**
setup wallet again **Check**
still should tell you about closed beta **Check, but same flashing/disappearing mentioned in previous step**
have your user name placed on the list (ask me, zach, or howard) **Check**
after your username is on the list, just open the wallet
you should see 10,000 hfc **Check, now to flush away the 10K!**
image

reset your wallet again **Check :(**
image

you should not get another 10K HFC - it shows the closed beta stuff like before **Check, see image above**
🍻 **This step is pending, need to grab some from my local! 💯**


Solid **Pass** as long as the flashing noticed and mentioned is not an issue :)

**Comment by davidkelly** [comment]
cool -- we will address the flash in the next PR that touches the wallet QML in a couple days 👍

**Pull request merged**